R/aalen_johansen.R

```diff
@@ -0,0 +1,114 @@
+#' Aalen Johansen Estimator
+#'
+#' This function calculates the Aalen Johansen estimator of
+#' adverse events  observed in `[0, tau]`.
+#' Please also refer to  \insertCite{stegherr_meta_analytic_2021;textual}{savvyr}.
+#'
+#' Estimating and comparing adverse event probabilities in the presence of varying follow-up times and competing events
+#' Regina Stegherr, Claudia Schmoor, Michael Lübbert, Tim Friede, Jan Beyersmann
+#' First published: 18 May 2021 https://doi.org/10.1002/pst.2130
+#'
+#' @typed data: data.frame
+#'  with columns including
+#'  - `time_to_event`: Time to the first AE, death or soft competing event.
+#'  - `type_of_event`: 0 for censored, 1 for AE, 2 for death, 3 for soft competing event.
+#'
+#' @typed tau: number
+#'  milestone at which Aalen-Johansen is computed.
+#'
+#' @typedreturn vector
+#'  with the following entries:
+#'
+#' - `ae_prob`: Estimated probability of AE.
+#' - `ae_prob_var`: Variance of that estimate.
+#' - `ce_prob`: Estimated probability of competing events.
+#' - `ce_prob_var`: Variance of competing events.
+#'
+#' @export
+#'
+#' @references
+#' \insertRef{stegherr_meta_analytic_2021}{savvyr}
+#'
+#' @examples
+#' set.seed(123)
+#' dat <- generate_data(n = 5, cens = c(2, 5), haz_ae = 2, haz_death = 3, haz_soft = 5)
+#' one_minus_kaplan_meier(dat, tau = 4)
```

Comment:
need to adapt the example to this function

Reply:
done